### ISSUE TITLE
Can reload bullets directly into magazine while it's in the gun

### ISSUE THREAD

**Issue by wapcaplet** [CleverRaven/Cataclysm-DDA#46151]
# Describe the bug

Given a gun with a removable magazine, using `r`eload on the gun allows reloading bullets directly into the magazine, while it's in the gun.

# Steps To Reproduce

1. Get a gun, compatible magazine, and some bullets
2. Make the magazine partially empty, then load it into the gun
3. Select the gun and `r`eload - ammo is inserted directly into the installed magazine

# Expected behavior

It should be necessary to remove the magazine from the gun before reloading bullets into it.

# Screenshots

Here I have a Glock 21 with a 13-round magazine, with only 2 bullets left. I have 5 more bullets in my inventory:

![image](https://user-images.githubusercontent.com/238395/102694384-dca04180-41dd-11eb-92c2-1781f6e82ba9.png)

When I `r`eload the gun, it puts bullets directly into the magazine:

![image](https://user-images.githubusercontent.com/238395/102694407-16714800-41de-11eb-831a-931de53ddb68.png)

# Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Pop; Description: Pop!_OS 20.04 LTS; Release: 20.04; Codename: focal; 
- Game Version: 0.E-8292-g00db780b75 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]

# Additional context

This is closely related to the complementary bug #46122 which prohibits reloading the entire magazine from a gun, unless you remove the current magazine first. They may have the same cause, but the symptoms and use cases are different.

**Comment by ToxiClay** [comment]
Nah, just reload the rounds straight through the ejection port. 🙃

**Comment by Svankensen** [comment]
I wouldn't mind this in particular if it takes longer than reloading a magazine that isn't inside a gun. As in, abstractly, you are taking the magazine out, and putting bullets in it. It makes for more manageable extended gunfights.

**Comment by Fosheze** [comment]
> I wouldn't mind this in particular if it takes longer than reloading a magazine that isn't inside a gun. As in, abstractly, you are taking the magazine out, and putting bullets in it. It makes for more manageable extended gunfights.

I 100% agree with this. While you shouldn't be able to reload a magazine while it is in the gun, making the player manually remove it all the time is adding a lot of keypresses for something that they have to do fairly frequently. If instead the time to reload a magazine while it is in the gun was simply increased by the amount of time it would take to remove it from the gun and put it back into the gun, that would have the same net effect while not adding additional keypresses to the reload action.

**Comment by Fosheze** [comment]
> > I wouldn't mind this in particular if it takes longer than reloading a magazine that isn't inside a gun.
> 
> > making the player manually remove it all the time is adding a lot of keypresses for something that they have to do fairly frequently.
> 
> I completely agree, but this QoL idea should be implemented as a feature, not as the side-effect of a bug.

I see your point however this bug has been a part of the game for at least as long as I have been playing so it is expected behavior for many players myself included. Outright removing the ability to reload a magazine that is in a gun is likely going to confuse many players who will just be left wondering why they can't reload their weapons anymore. It isn't intuitive that you need to remove the magazine first when that has not been case in the past.

Couldn't the already existing pocket access time system just be reused for reloading magazines. If I'm understanding correctly then a guns magazine well is already a pocket so all that should need to be done is to make the reloading action take that pocket access time into account and then the bug is fixed while preserving current functionality. This is entirely speculation of course but that was my thinking behind my original comment. If current functionality can be preserved while fixing this bug then it should.

**Comment by Xaleth** [comment]
Reloading bullets directly into the gun may take more moves?
